numpy/_core/src/multiarray/stringdtype/dtype.c

```diff
@@ -459,6 +460,18 @@ compare(void *a, void *b, void *arr)
     return ret;
 }
 
+int
+_sort_compare(const void *a, const void *b, void *context)
+{
+    PyArrayMethod_Context *sort_context = (PyArrayMethod_Context *)context;
+    PyArray_StringDTypeObject *sdescr =
+        (PyArray_StringDTypeObject *)sort_context->descriptors[0];
+
+    int ret = _compare((void *)a, (void *)b, sdescr, sdescr);
+    return ret;
+}
+
+// We assume the allocator mutex is already held.
```

Comment:
Good to add the comment! Nitpicky suggestion - does it make sense to define `_sort_compare` after this, given that it needs `_compare` but not vice versa? Or even together with `stringdtype_wrap_sort_loop` (in that case, repeat the comment...)

Reply:
I've moved it below `_compare` - thanks!